test/main/deepEqual.js

```diff
@@ -1581,6 +1581,35 @@ QUnit.test( "Test that must be done at the end because they extend some primitiv
 	}
 );
 
+QUnit.test( "Compare primitive objects", function( assert ) {
+	var SafeNumber = Number,
+	SafeString = String,
+	SafeBoolean = Boolean;
+
+	assert.equal( QUnit.equiv(new SafeNumber(1), new SafeNumber(1) ), true,
+		"Number: When create same parameter should be equal."
+	);
```

Comment:
Also: the [code style](http://contribute.jquery.org/style-guide/js/) requires a space before and after the arguments list, so `new SafeNumber(1)` => `new SafeNumber( 1 )`